django/src/rdwatch/views/site_image.py

```diff
@@ -146,3 +151,71 @@ def site_images(request: HttpRequest, id: UUID4):
     output['evaluationGeoJSON'] = site_eval_data['json']['evaluationGeoJSON']
     output['evaluationBBox'] = site_eval_data['json']['evaluationBBox']
     return output
+
+
+@router.post('/{id}/image_embedding', response=UUID4)
+def post_image_embedding(request: HttpRequest, id: int):
+    if not SiteImage.objects.filter(pk=id).exists():
+        raise Http404()
+    else:
+        return generate_image_embedding.delay(id).id
+
+
+@router.get('/{id}/image_embedding_status/{uuid}')
+def get_image_embedding_status(request: HttpRequest, id: int, uuid: UUID4):
+    if not SiteImage.objects.filter(pk=id).exists():
+        raise Http404()
+    else:
+        task = AsyncResult(uuid)
+        result = {
+            'state': task.state,
+            'status': task.status,
+        }
+        return result
+
+
+@router.get('/{id}/image', response=SiteImageSchema)
+def get_site_image(request: HttpRequest, id: int):
+    try:
+        site_image = (
+            SiteImage.objects.filter(pk=id)
+            .annotate(
+                timestamp_epoch=ExtractEpoch('timestamp'),
+                bbox=BoundingBox('image_bbox'),
+            )
+            .values(
+                'timestamp_epoch',
+                'source',
+                'cloudcover',
+                'image',
+                'observation_id',
+                'bbox',
+                'percent_black',
+                'image_dimensions',
+                'aws_location',
+                'image_embedding',
+                'pk',  # Assuming 'pk' is the primary key field name
+            )
+            .first()
+        )
+    except SiteImage.DoesNotExist:
+        raise Http404()
+
+    if site_image:
```

Comment:
Isn't this check redundant, as we've already raised a `Http404` by this point if the `site_image` doesn't exist?